framework/platform/bpl/uci/db/bpl_db.cpp

```diff
@@ -17,45 +18,94 @@
 namespace beerocks {
 namespace bpl {
 
+#define DB_FILE "prplMesh_db"
+
 bool db_has_entry(const std::string &entry_type, const std::string &entry_name)
 {
-    LOG(TRACE) << " - NOT IMPLEMENTED!";
-    return false;
+    LOG(TRACE) << entry_type << ":" << entry_name;
+    return uci_entry_exists(DB_FILE, entry_type, entry_name);
 }
 
 bool db_add_entry(const std::string &entry_type, const std::string &entry_name,
                   const std::unordered_map<std::string, std::string> &params)
 {
-    LOG(TRACE) << " - NOT IMPLEMENTED!";
-    return true;
+    LOG(TRACE) << entry_type << ":" << entry_name;
+
+    // Check if entry of the same name (and any type) exist.
+    if (!uci_entry_exists(DB_FILE, "", entry_name)) {
+        LOG(DEBUG) << "Entry " << entry_name << " does not exist! Create new one.";
+        if (!uci_add_entry(DB_FILE, entry_type, entry_name)) {
+            LOG(ERROR) << "Failed to create entry " << entry_name << "!";
+            return false;
+        }
+    } else {
+        // Entry exists, validate type match
+        if (uci_entry_exists(DB_FILE, entry_type, entry_name)) {
+            LOG(ERROR) << "Entry " << entry_name << " exists, but under diffrent type!";
+            return false;
+        }
+    }
+    // Update the new/existing entry
+    LOG(DEBUG) << "Update entry " << entry_name << " with new values.";
+    return uci_set_entry(DB_FILE, entry_type, entry_name, params);
 }
 
 bool db_set_entry(const std::string &entry_type, const std::string &entry_name,
                   const std::unordered_map<std::string, std::string> &params)
 {
-    LOG(TRACE) << " - NOT IMPLEMENTED!";
-    return true;
+    LOG(TRACE) << entry_type << ":" << entry_name;
+
+    if (!uci_entry_exists(DB_FILE, entry_type, entry_name)) {
+        LOG(ERROR) << "Entry " << entry_name << " not found, or exists under a diffrent type!";
+        return false;
+    }
+    // Update the new/existing entry
+    LOG(DEBUG) << "Update entry " << entry_name << " with new values.";
+    return uci_set_entry(DB_FILE, entry_type, entry_name, params);
 }
 
 bool db_get_entry(const std::string &entry_type, const std::string &entry_name,
```

Comment:
shouldn't it be `db_get_entry_by_name`

Reply:
No since you either get a single entry by it's unique identifier (name)
or get all entries with the same type

Reply:
@itayx this is not true
if the API is called with entry_type **set** and entry_name **not-set** and params **set**
you will get the params from all clients - what is a valid use-case for this?

you need to verify that the name is **always** provided:
```
if (entry_name.empty()) {
    LOG(ERROR)
    return false;
}
```
(in the bpl_db layer you need to think from a bpl_db perspective and not from UCI perspective)